### PR TITLE
DirectPipelineRunner bug fixes

### PR DESCRIPTION
**Pull request by aaltay**
- Execute empty [] | pipelines to the end.
- use pickler to serialize/deserialize DoFns instead of deepcopy similar to the othe execution environments.

### CODE REVIEW

**Comment by aaltay** [comment]
@ccy Please review.

File: sdks/python/apache_beam/runners/direct/transform_evaluator.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ def start_bundle(self):
 
     self._counter_factory = counters.CounterFactory()
 
-    dofn = copy.deepcopy(transform.dofn)
+    dofn = pickler.loads(pickler.dumps(transform.dofn))
```
Comment (charlesccychen) [inline sdks/python/apache_beam/runners/direct/transform_evaluator.py]: Can you add a todo here to perhaps store this in serialized form?  This seems suboptimal but all right for now and not yet worth changing.

**Comment by charlesccychen** [comment]
Thanks! LGTM.

**Comment by robertwb** [comment]
Thanks. Ping me to merge once comment is addressed and tests are green.

**Comment by aaltay** [comment]
Thank you for the quick review.

@robertwb this is ready for merging. (comment addressed, tests passed.

**Comment by aaltay** [comment]
Thank you.

**Pull request closed without merging**